spec/models/alchemy/page_spec.rb

```diff
@@ -338,6 +338,14 @@ module Alchemy
           expect(news_page.elements.pluck(:name)).to include('contactform')
         end
       end
+
+      context 'destruction' do
+        let!(:page) { create(:alchemy_page, autogenerate_elements: true) }
+
+        it 'destroys elements along with itself' do
+          expect { page.destroy! }.to change(Alchemy::Element, :count)
```

Comment:
I think it makes sense to explicitly test the absolute changed numbers here, to avoid false positives from potential side effects.